be/src/io/compressed_input_stream.h

```diff
@@ -81,4 +82,26 @@ class CompressedInputStream final : public InputStream {
     bool _stream_end = false;
 };
 
+class CompressedSeekableInputStream final : public SeekableInputStream {
+public:
+    CompressedSeekableInputStream(std::shared_ptr<CompressedInputStream> source) : _source(source) {}
+
+    StatusOr<int64_t> read(void* data, int64_t size) override { return _source->read(data, size); }
+
+    Status skip(int64_t n) override { return _source->skip(n); }
+
+    bool allows_peek() const override { return _source->allows_peek(); }
+
+    StatusOr<std::unique_ptr<NumericStatistics>> get_numeric_statistics() override {
+        return _source->get_numeric_statistics();
+    }
+
+    Status seek(int64_t position) override { return Status::NotSupported(""); }
```

Comment:
`CompressedSeekableInputStream` is a `SeekableInputStream` but does not support `seek()`?

Reply:
Yes. The main purpose of it, is to serve as underlying stream for RandomAccessFile.

You know there are some classes which use RandomAccessFile as underlying file handler, but sometimes also want to access compressed file in sequential way, which means it has to use SequenceAccessFile.

Reply:
> CompressedSeekableInputStream

It's like a ducking type to make compiler happy.

Reply:
@sduzh What do you think?